tests/unit/iontests.js

```diff
@@ -11,126 +11,365 @@
+            'bad/stringWithLatinEncoding.10n',
+            'bad/structOrderedEmpty.10n',
+            'bad/annotationLengthTooShortScalar.10n',
+            'bad/annotationLengthTooShortContainer.10n',
+            'bad/annotationNested.10n',
+            'bad/timestamp/timestampHourWithoutMinute.10n',
+
+
+
+        ];
+
+        // For debugging, put single files in this list to have the test run only
+        // that file.   (But don't forget to clean this up on checkin!)
+        let debugList = [
+            //'good/sexps.ion'
+        ];
+
+        let goodUnskipped = [];
+        for (let path of goodAccumulator) {
+            let spath = path.replace(/\\/g, "/");
+            let shouldSkip = false;
+            for (let skip of skipList) {
+                if (spath.endsWith(skip)) {
+                    shouldSkip = true;
+                    break;
+                }
+            }
+            if (debugList.length > 0) {
+                shouldSkip = true;
+                for (let debug of debugList) {
+                    if (spath.endsWith(debug)) {
+                        shouldSkip = false;
+                        break;
+                    }
+                }
+            }
+            if (!shouldSkip) {
+                goodUnskipped.push(path);
+            }
         }
-      }
-      if (!shouldSkip) {
-        unskipped.push(path);
-      }
-    }
 
-    var suite = {
-      name: 'Good tests',
-    };
-
-    function exhaust(reader) {
-      var tries = 0;
-      for (;;) {
-        // Safety valve
-        tries++;
-        if (tries > 1000) {
-          throw new Error("Reader spinning forever!");
+        let badUnskipped = [];
+        for (let path of badAccumulator) {
+            let spath = path.replace(/\\/g, "/");
+            let shouldSkip = false;
+            for (let skip of skipList) {
+                if (spath.endsWith(skip)) {
+                    shouldSkip = true;
+                    break;
+                }
+            }
+            if (debugList.length > 0) {
+                shouldSkip = true;
+                for (let debug of debugList) {
+                    if (spath.endsWith(debug)) {
+                        shouldSkip = false;
+                        break;
+                    }
+                }
+            }
+            if (!shouldSkip) {
+                badUnskipped.push(path);
+            }
+        }
+
+        let goodSuite = {
+            name: 'Good tests'
+        };
+        let badSuite = {
+            name: 'Bad tests'
+        };
+
+        let eventStreamSuite = {
+            name: 'EventStream tests'
+        };
+        let badEventStreamSuite = {
+            name: 'Bad EventStream tests'
+        };
+
+        function goodExhaust(reader) {
+            let tries = 0;
+            for (;;) {
+                tries++;
+                if (tries > 1000) {
+                    throw new Error("Reader spinning forever!");
```

Comment:
How do you know it won't halt??!

In all seriousness, I'd remove this. If the reader gets caught in an infinite loop, that's a pretty urgent issue that should be addressed as it is found.